lib/bamboo_info_provider.rb

```diff
@@ -8,18 +9,19 @@ class BambooInfoProvider
   def initialize(subdomain)
     @subdomain = subdomain
     @api_key = ENV["BAMBOO_HR_API_KEY"]
-    if @api_key.nil?
-      raise "Invalid API key! Have you set one?"
-    end
+    raise "Invalid API key! Have you set one?" if @api_key.nil?
+    @cache = BambooCache.new
   end
 
   def employees
     data = JSON.parse(employees_data.body)
```

Comment:
You could have here a method named `get_employees` that will return `JSON.parse(employes_data.body)["employees"]`